### PR TITLE
feat: add HTPipeFinalAuthorize

### PR DESCRIPTION
**Pull request by VolodymyrWork23**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -184,6 +186,185 @@ export function HTPipeAttachData<
...
+    ? PromiseResolveOrSync<ReturnType<TRight['finalAuthorize']>>
+    : never
+>(left: TLeft, right: TRight) {
+  if (isHasFinalAuthorize(left) && isHasFinalAuthorize(right)) {
+    return {
+      finalAuthorize: async (
+        context: TContextOutLeft extends TContextInRight
+          ? TContextInLeft
+          : TContextInRight & TContextInLeft
+      ) => {
```
Comment (captaincaius) [inline src/index.ts:341]: this seems different than the other one - for example, the || false.  why?
- [ ] make it similar to preAuthorize

File: src/index.ts [hunk 2]
```diff
@@ -184,6 +186,185 @@ export function HTPipeAttachData<
...
+          (await Promise.resolve(left.finalAuthorize(context))) || false;
+        const leftPassed = authorizationPassed(leftOut);
+        if (!leftPassed) {
+          return false;
+        }
+        const leftContextOut = leftOut === true ? {} : leftOut;
+        const rightIn = {
+          ...context,
+          ...leftContextOut,
+        };
```
Comment (captaincaius) [inline src/index.ts:352]: - [ ] same as in the other one - just return rightOut

File: src/index.ts (hunk 1 above)
Comment (VolodymyrWork23) [reply src/index.ts:341]: done

File: src/index.ts (hunk 2 above)
Comment (VolodymyrWork23) [reply src/index.ts:351]: done

**Comment by captaincaius** [comment]
- [ ] resolve conflict
- [ ] in a SEPARATE commit, refactor HTPipePre to use the new authorizePassed thing

**Pull request merged**